ui/package.json

```diff
@@ -27,7 +27,7 @@
         "json-merge-patch": "^0.2.3",
         "moment": "^2.29.4",
         "moment-timezone": "^0.5.43",
-        "monaco-editor": "0.20.0",
+        "monaco-editor": "^0.30.0",
```

Comment:
What about `react-monaco-editor`? https://github.com/argoproj/argo-workflows/pull/11598?

Reply:
Oh I hadn't seen that one, you had only mentioned #11475 in the meeting. Upgrading `monaco-editor` didn't seem to require updating `react-monaco-editor` in tandem, but I can take a look at the error there too now that I know of it

Reply:
Oh that Snyk PR was literally opened a few hours ago just before you commented here 😅 
Was that a manual run?

Reply:
Hmm build seems unrelated for that one. And even the version of `react-monaco-editor` that we're on, 0.50.1, actually requires a newer version of `monaco-editor`, [^0.34.0](https://github.com/react-monaco-editor/react-monaco-editor/blob/0f119c8e805d489ed35616b90af595dad32a7709/package.json#L53). 
That Snyk PR upgrades to 0.54.0, which requires `monaco-editor` [^0.39.0](https://github.com/react-monaco-editor/react-monaco-editor/blob/6ab247bc9e17271b43e27c729902def77ef1009c/package.json#L53).

So er, the current set-up already uses incompatible peerDeps, but this one brings it at least closer 😅

Reply:
We might want to make CI UI install fail on invalid peerDeps. Though right now all the React versioning is incompatible too, but still working (as an unsupported peerDep doesn't _necessarily_ mean it's incompatible, but means it probably hasn't been tested at least)

Reply:
`react-monaco-editor` seems to be mostly pass through, as the `options` prop was directly impacted by the `monaco-editor` version

Reply:
Investigated `react-monaco-editor` in https://github.com/argoproj/argo-workflows/pull/11598#issuecomment-1696429512